### ISSUE TITLE
Add to cart button is missing `data-price` attribute used by various trackers.

### ISSUE THREAD

**Issue by message-dimke** [woocommerce/woocommerce-gutenberg-products-block#11116]
## Describe the bug

TLDR; When using Products (Beta), Product by Category or Products by Attribute blocks, Pinterest AddToCart event does not receive a value parameter value when sending an event to Pinterest.

Please, follow the link below for more details.

[Pinterest Tag AddToCart event is missing value attribute value when using blocks.](https://github.com/woocommerce/pinterest-for-woocommerce/issues/761)

## To reproduce

- Add a page with a Products (Beta), Product by Category or Products by Attribute block on it.
- Install Pinterest Tag Helper extension into your browser (I use Chrome for that matter).
- Navigate to the page with the added block and click Add to Cart button.4.
- See an AddToCart event inside Pinterest Tag Helper has no value attribute value (it has it set to null, instead of a product price).
